pkg/query/storeset.go

```diff
@@ -421,6 +432,7 @@ func (s *StoreSet) cleanUpStoreStatuses() {
 	for addr, status := range s.storeStatuses {
 		if _, ok := s.stores[addr]; !ok {
 			if now.Sub(status.LastCheck) >= s.unhealthyStoreTimeout {
+				s.storeNodeStatus.DeleteLabelValues(addr)
```

Comment:
Not sure, how Prometheus does it? I think it keeps the thing forever AFAIK. Again, let's separate UI with metric system

Reply:
Prometheus's metrics for the Alertmaanger are the closest thing to this, looks like we aren't doing any cleanup there currently which should be fixed.

Reply:
TODO: Add issue on AM